content/include/mock_assert_example.md

```diff
@@ -0,0 +1,33 @@
+    // ...
+    t.Run("nominal case", func(t *testing.T) {
+        t.Parallel()
+
+        mock := mocks.BaselineTransport(t)
+        mock.Request = func(method, url string) (*http.Response, error) {
+            assert.Equal(t, http.MethodPost, method)
+            assert.Equal(t, "https://google.com", url)
```

Comment:
Should this be require? It should stop here right?

Reply:
No, I don't see why this should be a `require` 🤔 We're just checking that we called the mock with the right parameters. The only case where we want to use require is when we make an assertion which if it fails, will for sure cause others to fail or cause panics down the line. For example if we're making assertions on `x.AttributeY` and `x.AttributeZ` we probably want to first do `require.NotNil(t, x)`. If we're making assertions on something returned along with an error, we probably want to do `require.NoError(t, err)` before making assertions on that thing. If we're making setup calls we want to require for them not to error.

Does that make sense? Maybe I should add another note to clarify when to use assert and when require then 😄

Reply:
No need, I thought you wanted to fail the test in the spot when some of the arguments don't match. I think that is how gomock works.